src/main/java/com/jcabi/maven/plugin/AjcMojo.java

```diff
@@ -536,12 +543,27 @@ private boolean hasSourceroots() {
      */
     private void copyUnwovenClasses()
         throws MojoFailureException {
-        this.unwovenClassesDir.mkdirs();
-        Logger.info(
-            this, "Unwoven classes will be copied to %s",
-            this.unwovenClassesDir
-        );
         if (this.hasClasses()) {
+            final String phase = this.execution.getLifecyclePhase();
+            if ("process-classes".equals(phase)) {
+                this.unwovenClassesDir.mkdirs();
+                Logger.info(
+                    this, "Unwoven classes will be copied to %s",
+                    this.unwovenClassesDir
+                );
+            } else if ("process-test-classes".equals(phase)) {
+                final String testsuf = "-test";
+                final StringBuilder fpb = new StringBuilder();
+                fpb.append(this.unwovenClassesDir.getPath()).append(testsuf);
+                this.unwovenClassesDir = new File(fpb.toString());
+                this.unwovenClassesDir.mkdirs();
+                Logger.info(
+                    this, "Unwoven test classes will be copied to %s",
+                    this.unwovenClassesDir
+                );
+            } else {
+                return;
```

Comment:
@amihaiemil breaking method execution somewhere in the middle is not what we do... so you see some other option?